game-core/src/test/java/games/strategy/triplea/delegate/battle/steps/fire/general/FiringGroupSplitterGeneralTest.java

```diff
@@ -210,7 +283,7 @@ void excludeSuicideOnAttackTargetsIfDefense() {
     final Unit fireUnit = givenAnyUnit();
 
     final List<FiringGroup> firingGroups =
-        FiringGroupSplitterGeneral.of(DEFENSE, unit -> true, "")
```

Comment:
Looking at these usage examples, and in the prod code, it looks like the update to a flag instead of passing a predicate was a strong update, it is a much nicer looking API now :+1: